### PR TITLE
Add 'audio_delay' FX

### PR DESCRIPTION
**Pull request by mondeja**
This is a common effect working with audio. Keep in mind that this effect does not fit a `loop` FX behaviour, since the sound is multidimensional because you can have multiples copies playing at the same time. Even with an `offset` attribute in current `loop` FX this effect would be impossible to achieve.

- [ ] I have provided code that clearly demonstrates the bug and that only works correctly when applying this fix
- [x] I have added suitable tests demonstrating a fixed bug or new/changed feature to the test suite in `tests/`
- [x] I have properly documented new or changed features in the documentation or in the docstrings
- [x] I have properly explained unusual or unexpected code in the comments around it
- [x] I have formatted my code using `black -t py36`

### CODE REVIEW

File: moviepy/audio/fx/audio_delay.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    decay (float)
+      Multiplication factor for the volume level of the last repetition. Each
+      repetition will have a value in the linear function between 1 and this value,
+      increasing or decreasing constantly. Keep in mind that the last repetition
+      will be muted if this is 0, and if is greater than 1, the volume will increase
+      for each repetition.
+
+    Examples
+    ========
```
Comment (tburrows13) [inline moviepy/audio/fx/audio_delay.py:30]:
```suggestion
    offset : float, optional
      Gap between repetitions start times, in seconds.

    n_repeats : int, optional
      Number of repetitions (without including the clip itself).

    decay : float, optional
      Multiplication factor for the volume level of the last repetition. Each
      repetition will have a value in the linear function between 1 and this value,
      increasing or decreasing constantly. Keep in mind that the last repetition
      will be muted if this is 0, and if is greater than 1, the volume will increase
      for each repetition.

    Examples
    --------
```

According to https://numpydoc.readthedocs.io/en/latest/format.html#sections, this is the preferred formatting style.

**Comment by tburrows13** [comment]
I'm not sure about this one... isn't it basically the same as #1358? I wonder if it would make more sense to implement this one in `loop`, if you think that the `decay` would work there?

**Comment by mondeja** [comment]
> I'm not sure about this one... isn't it basically the same as #1358? I wonder if it would make more sense to implement this one in loop, if you think that the decay would work there?

Keep in mind that is not the same as a loop. In a loop, the whole loop is played and after that starts another copy. In a delay effect, you can have multiple copies running at the same time.

**Comment by tburrows13** [comment]
> Keep in mind that is not the same as a loop. In a loop, a whole clip is played and after that starts another copy. In a delay effect, you can have multiple copies playing at the same time.

Oh right, in which case this is fine. There would be no way to implement loop with #1358 that allows a 'negative' gap, so this is a good way to do it.

I have a feeling that the description or name could be improved, but I can't take that further right now. Depending on how those change, perhaps the default decay should be 1?

**Comment by mondeja** [comment]
> I have a feeling that the description or name could be improved

The description sure, but the name is a standard in music and sound areas, it feel to me that is the right name.

**Comment by mondeja** [comment]
Do you see the description right now @tburrows13?

**Pull request merged**